language/types_values_variables.md

```diff
@@ -456,6 +461,333 @@ For conversion from `String` both float and integer formats are supported.
 * In `String` format, integer prefixes for hex and binary radix are understood (but not octal since
   floating point in string format may start with a '0').
 
+### Timespan ([from, to])
+
+Since version 4.8.0
+
+Represents a range of timespan values. The default is the range +/- Infinity.
+
+A timespan is an duration, measured in seconds, with nano-second precision.
+
+A `Timespan` range behaves as an `Integer` range and accepts integer, float, or timespan values when
+specifying the range. It is however, not possible to iterate over a `Timespan` range.
+
+#### Type Algebra on Timespan
+
+    Timespan ∪ Timespan            → Timespan
+    Timespan ∪ Integer             → Numeric
+    Timespan ∪ Numeric             → Numeric
+    Timespan ∪ (T ∈ Scalar)        → Scalar
+    Timespan ∪ (T ∉ Scalar)        → Any
+    Timespan[a, b] ∪ Timespan[c, d]   → Timespan[min(a, c), max(b,d)]
+
+#### Timespan.new
+
+A new `Timespan` can be created from `Integer`, `Float`, `String`, and `Hash` values. Several variants of the constructor are provided.
+
+##### Timespan from seconds
+
+When a Float is used, the decimal part represents fractions of a second.
+
+```puppet
+function Timespan.new(
+  Variant[Float, Integer] $value
+)
+```
+
+##### Timespan from days, hours, mintues, seconds, and fractions of a second
+
+The arguments can be passed separately in which case the first four, days, hours, minutes, and seconds are mandatory and the rest are optional.
```

Comment:
The `first four;` (now it looks like the word `four` is part of the enumeration, use a semicolon to make the enumeration an explanation of what "the first four" means).